src/graphql/event/tls.rs

```diff
@@ -182,3 +182,181 @@ impl From<database::BlockListTls> for BlockListTls {
         Self { inner }
     }
 }
+
+pub(super) struct SuspiciousTlsTraffic {
+    inner: database::SuspiciousTlsTraffic,
+}
+
+#[Object]
+impl SuspiciousTlsTraffic {
+    async fn time(&self) -> DateTime<Utc> {
+        self.inner.time
+    }
+
+    async fn source(&self) -> &str {
+        &self.inner.source
+    }
+
+    async fn src_addr(&self) -> String {
+        self.inner.src_addr.to_string()
+    }
+
+    /// The two-letter country code of the source IP address. `"XX"` if the
+    /// location of the address is not known, and `"ZZ"` if the location
+    /// database is unavailable.
+    async fn src_country(&self, ctx: &Context<'_>) -> String {
+        country_code(ctx, self.inner.src_addr)
+    }
+
+    async fn src_customer(&self, ctx: &Context<'_>) -> Result<Option<Customer>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.customer_map();
+        find_ip_customer(&map, self.inner.src_addr)
+    }
+
+    async fn src_network(&self, ctx: &Context<'_>) -> Result<Option<Network>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.network_map();
+        find_ip_network(&map, self.inner.src_addr)
+    }
+
+    async fn src_port(&self) -> u16 {
+        self.inner.src_port
+    }
+
+    async fn dst_addr(&self) -> String {
+        self.inner.dst_addr.to_string()
+    }
+
+    /// The two-letter country code of the destination IP address. `"XX"` if the
+    /// location of the address is not known, and `"ZZ"` if the location
+    /// database is unavailable.
+    async fn dst_country(&self, ctx: &Context<'_>) -> String {
+        country_code(ctx, self.inner.dst_addr)
+    }
+
+    async fn dst_customer(&self, ctx: &Context<'_>) -> Result<Option<Customer>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.customer_map();
+        find_ip_customer(&map, self.inner.dst_addr)
+    }
+
+    async fn dst_network(&self, ctx: &Context<'_>) -> Result<Option<Network>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.network_map();
+        find_ip_network(&map, self.inner.dst_addr)
+    }
+
+    async fn dst_port(&self) -> u16 {
+        self.inner.dst_port
+    }
+
+    async fn proto(&self) -> u8 {
+        self.inner.proto
+    }
+
+    async fn last_time(&self) -> i64 {
```

Comment:
This should be `StringNumber`. It needs to be applied to similar cases as well.

Reply:
Done